src/breaking-change.ts

```diff
@@ -162,13 +165,183 @@ async function runOad(oldSpec: string, newSpec: string) {
+    return glob.sync(pathPattern, {
+      ignore: [
+        "**/examples/**/*.json",
+        "**/quickstart-templates/*.json",
+        "**/schema/*.json",
+      ],
+    });
+  }
+
+  getVersionMapping(swaggerFile: string) {
+    const swaggerMetaData: SwaggerMetaData[] = [];
+    const folder = this.getRPFolder(swaggerFile);
+    if (!folder) {
+      return swaggerMetaData
+    }
+    const allSwaggers = this.getAllSwaggers(folder);
+    for (const swagger of allSwaggers) {
+      const version = getVersionFromInputFile(swagger);
+      const fileName = path.basename(swagger);
+      // Needs to consider other cases
+      const versionType = swagger.includes("preview") ? "preview" : "stable";
+      swaggerMetaData.push({
+        version,
+        versionType,
+        fileName,
+        folder,
+      });
+    }
+    return swaggerMetaData;
+  }
+
+  getClosestVersion(swaggerFile: string, type: SwaggerVersionType) {
+    const versions = this.getVersionMapping(swaggerFile);
+    const fileName = path.basename(swaggerFile)
+    const currentVersion = getVersionFromInputFile(swaggerFile)
+    try {
+      const version = versions
+        .filter((v) => v.fileName === fileName && v.versionType === type && v.version !== currentVersion)
+        .reduce((previous, current) =>
+          previous.version > current.version ? previous : current
+        );
+        return path.join(version.folder,version.versionType,version.version,version.fileName)
+    } catch (e) {
+      return undefined;
+    }
+  }
+
+  getClosestPreview(swaggerFile: string) {
+    return this.getClosestVersion(swaggerFile, "preview");
+  }
+
+  getClosestStale(swaggerFile: string) {
+    return this.getClosestVersion(swaggerFile, "stable");
+  }
+}
+
+export class CrossVersionBreakingDetector {
+  swaggers :string[]= []
+  pr : PullRequestProperties
+  versionManager: SwaggerVersionManager = new SwaggerVersionManager()
+  constructor(pullRequest: PullRequestProperties,newSwaggers:string[]) {
+    this.swaggers = newSwaggers
+    this.pr = pullRequest
+  }
+
+  async diffOne(oldSpec: string ,newSpec : string) {
+    try {
+      await runOad(path.resolve(this.pr!.workingDir,oldSpec ), newSpec);
+    }
+    catch(e) {
+      const errors = []
+      errors.push({
+        error: e,
+        old: targetHref(
+          utils.getRelativeSwaggerPathToRepo(
+            path.resolve(this.pr!.workingDir, oldSpec)
+          )
+        ),
+        new: blobHref(utils.getRelativeSwaggerPathToRepo(newSpec)),
+      });
+      appendException(errors)
+    }
+  }
+
+  async getBreakingChangeBaseOnPreviewVersion() {
+    for (const swagger of this.swaggers) {
+      const previous = await utils.doOnTargetBranch(this.pr, async () => {
+        return this.versionManager.getClosestPreview(swagger)
+      })
+      if (previous) {
+        await this.diffOne(path.resolve(this.pr!.workingDir,previous), swagger);
+      }
+    }
+  }
+
+  async getBreakingChangeBaseOnStableVersion() {
```

Comment:
naming. get => detect.  get means return something.

Reply:
Oh, thanks , fixed it